pkg/scheduler/framework/statement.go

```diff
@@ -194,7 +196,117 @@ func (s *Statement) unpipeline(task *api.TaskInfo) error {
 	return nil
 }
 
-// Discard operation for evict and pipeline
+// Save the allocate operation on task
+func (s *Statement) Allocate(task *api.TaskInfo, hostname string) error {
```

Comment:
Most of code is same as `ssn.Allocate()`, could we deduplicate them, e.g. move same code to a new function?

Reply:
We will finally remove the Allocate func in allocate.go. For now there is still other places using it.